web/src/js/iconhandler.js

```diff
@@ -95,10 +94,32 @@ export default class IconHandler
             }
         }
     }));
+  }
+
+  async requestTxResult(txHash)
```

Comment:
Please modify function so that it's easier to understand and that only one toast is shown for each voting.
Right now, if we vote more than once, at each vote all previous toasts are also shown.